genesyscloud/resource_genesyscloud_processautomation_trigger.go

```diff
@@ -156,15 +157,38 @@ func resourceProcessAutomationTrigger() *schema.Resource {
 				Elem:        matchCriteria,
 			},
 			"event_ttl_seconds": {
-				Description: "How old an event can be to fire the trigger",
-				Type:        schema.TypeInt,
-				Optional:    true,
-				Required:    false,
+				Description:  "How old an event can be to fire the trigger. Must be an number greater than or equal to 10",
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateEventTTLGreaterThanOrEqualTo10(),
```

Comment:
+1

Reply:
Looks like you can avoid it being set to 0 by updating the code in the read function to
```
		if trigger.EventTTLSeconds != nil && *trigger.EventTTLSeconds > 0 {
			d.Set("event_ttl_seconds", *trigger.EventTTLSeconds)
		}
```
and omitting the else. I hadn't known that setting an attribute to nil sets it to 0 if it's an int. We might have to set if that's causing issues in other resources.

Reply:
@ronanwatkins doing that then gives a response that looks like this in the CLI.

`the_trigger = {
  "enabled" = true
  "event_ttl_seconds" = tonumber(null)
  "id" = "a0e6766d-bc63-4ea0-a7ef-aa14bd50e054"
  "match_criteria" = toset([
    {
      "json_path" = "mediaType"
      "operator" = "In"
      "value" = ""
      "values" = tolist([
        "CHAT",
      ])
    },
  ])
  "name" = "Test Trigger - terraform creation - abcd-1234-values"
  "target" = toset([
    {
      "id" = "ae1e0cde-875d-4d13-a498-615e7a9fe956"
      "type" = "Workflow"
    },
  ])
  "topic_name" = "v2.detail.events.conversation.{id}.customer.end"
}
`

'tonumber(null)' just seems like nonsense, and i think i would prefer the empty string approach we have implemented

Reply:
What are you doing to get this output from the CLI? I've never seen the CLI output any HCL like this

Reply:
I have this in my versions.tf:
```output "the_trigger" {
  description = "the test trigger"
  value       = resource.genesyscloud_processautomation_trigger.test-trigger
}```

Reply:
Really not sure how it prints that stuff.
This is my config:
```
resource "genesyscloud_processautomation_trigger" "test-trigger" {
  name       = "Test Trigger"
  topic_name = "v2.detail.events.conversation.{id}.customer.end"
  enabled    = true
  target {
    id   = genesyscloud_flow.workflow.id
    type = "Workflow"
  }
  match_criteria {
    json_path = "mediaType"
    operator  = "Equal"
    value     = "CHAT"
  }
}
```
And I got this output after updating it to remove `event_ttl_seconds`:
```
......
Terraform will perform the following actions:

  # genesyscloud_processautomation_trigger.test-trigger will be updated in-place
  ~ resource "genesyscloud_processautomation_trigger" "test-trigger" {
      - event_ttl_seconds = 60 -> null
        id                = "dff89de7-ccac-4814-85f9-080557fd79bb"
        name              = "Test Trigger"
        # (2 unchanged attributes hidden)


      + target {
          + id   = "bcceac36-ffd3-4843-a3f9-5a867fa0f657"
          + type = "Workflow"
        }
      - target {
          - id   = "fb79b041-c332-458d-87c9-62e9913d7d1c" -> null
          - type = "Workflow" -> null
        }
        # (1 unchanged block hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
genesyscloud_processautomation_trigger.test-trigger: Modifying... [id=dff89de7-ccac-4814-85f9-080557fd79bb]
genesyscloud_processautomation_trigger.test-trigger: Modifications complete after 2s [id=dff89de7-ccac-4814-85f9-080557fd79bb]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

Reply:
if i remove my output section then i get the same thing as you. i was just concerned that if someone using this provider adds the terraform output section, then they will get strange looking responses for event ttl

Reply:
Oh yeah, you used outputs. I wouldn't be too worried about what they see there.